client/app/services/query.js

```diff
@@ -253,6 +253,12 @@ export class Parameter {
       }
     }
 
+    if (this.type === 'query' && !isEmptyValue(value)) {
+      if (this.multiValuesOptions && !isArray(value) && value !== null) {
+        value = [value];
```

Comment:
QueryBasedDropdown with multi-values should always hold array values in its structure.